libcudacxx/include/cuda/std/__memory/is_sufficiently_aligned.h

```diff
@@ -33,7 +33,9 @@ _CCCL_BEGIN_NAMESPACE_CUDA_STD
 template <size_t _ByteAlignment, class _ElementType>
 [[nodiscard]] _CCCL_API inline bool is_sufficiently_aligned(_ElementType* __ptr) noexcept
 {
-  return ::cuda::std::bit_cast<uintptr_t>(__ptr) % _ByteAlignment == 0;
+  static_assert(::cuda::is_power_of_two(_ByteAlignment), "alignment must be a power of two");
+  static_assert(_ByteAlignment % sizeof(_ElementType) == 0, "element size must be a multiple of the alignment");
```

Comment:
Important: I don't think this is correct. I can ask whether an `int3*` is aligned to 16 and that should be ok.